### PR TITLE
User can search for posts

### PR DESCRIPTION
**Pull request by ravencovington**
[#95331120](https://www.pivotaltracker.com/story/show/95331120)

- Creates PostsSearchesController to handle solr/sunspot search
- Adds searchable block to Post model
- Rearranges header to include search box
- Adds a PostsSearches view and route

### CODE REVIEW

**Comment by wolfbrain** [comment]
:eyes: @srbiv

**Comment by wolfbrain** [comment]
:eyes: @pdougall1

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require "rails_helper"
+include SunspotMatchers
+
+RSpec.describe Post, type: :model do
+  describe "searchable fields" do
+    it { should have_searchable_field(:content) }
```
Comment (srbiv) [inline spec/models/post_spec.rb]: A more modern way to write this spec would be:
` it { is_expected.to have_searchable_field(:content) }`

[Rspec](https://relishapp.com/rspec/rspec-expectations/docs/syntax-configuration)'s docs recommend against using it.

Comment (ravencovington) [inline spec/models/post_spec.rb]: Thanks. I'll fix that.

**Comment by srbiv** [comment]
This looks good, although I would recommend using Postgres' built in full text search instead of Solr for doing simple full text searches like this as it doesn't introduce the extra dependency of Solr.

**Comment by ravencovington** [comment]
Thanks. That makes sense. I'll definitely look into the built in full text search.

**Pull request merged**